src/core/utilities/attribute-target.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  ElementRef,
+  Renderer,
+} from 'angular2/core';
+
+
+export interface AttributeTargetDirectiveInterface {
+  attributeMap: { [name: string]: any };
+}
+
+
+export class MdAttributeTarget<T extends AttributeTargetDirectiveInterface> {
+  public directive: T;
+  public element: ElementRef;
+  public renderer: Renderer;
+
+  ngAfterViewInit() {
```

Comment:
I don't believe this will keep the attributes up to date when the bindings change.